quinn/src/platform/unix.rs

```diff
@@ -375,8 +396,8 @@ fn prepare_msg(
                 IpAddr::V4(v4) => {
                     let pktinfo = libc::in_pktinfo {
                         ipi_ifindex: 0,
-                        ipi_spec_dst: unsafe {
-                            *(v4 as *const Ipv4Addr as *const () as *const libc::in_addr)
+                        ipi_spec_dst: libc::in_addr {
+                            s_addr: u32::from_ne_bytes(v4.octets()),
```

Comment:
I don't think `from_ne_bytes` is correct here. `s_addr` should be in network byte order. We should probably have unit tests for this code...

Reply:
> I don't think from_ne_bytes is correct here. s_addr should be in network byte order. 

`from_ne_bytes` takes the 4 bytes and copies them into the resulting `u32` without changing byte order. As the `octets()` function returns in network order, `from_ne_bytes` is what we need to use. There are tests for this code in fact. Try changing to `from_be_bytes` and some tests will fail :). If you are still not convinced, check the socket2 code, [it does the same](https://docs.rs/socket2/0.3.19/src/socket2/sockaddr.rs.html#206).